d4s2_api/utils.py

```diff
@@ -117,27 +122,78 @@ class ProcessedMessage(Message):
     def get_templates(self, delivery_details):
         return delivery_details.get_action_template_text(self.process_type)
 
-    def __init__(self, delivery, user, process_type, reason=''):
+    def __init__(self, delivery, user, process_type, reason='', warning_message=''):
         """
         Generates a Message to the sender reporting whether or not the recipient accepted the delivery
         """
         self.process_type = process_type
         super(ProcessedMessage, self).__init__(delivery, user, process_type=process_type, reason=reason,
-                                               direction=MessageDirection.ToSender)
+                                               direction=MessageDirection.ToSender,
+                                               warning_message=warning_message)
 
 
 def accept_delivery(delivery, user):
     """
-    Communicates with DukeDS via DDSUtil to accept the project transfer
+    Communicates with DukeDS via DDSUtil to accept the project transfer.
+    Also gives download permission to the users in the delivery's share_to_users list.
     :param user: The user with a DukeDS authentication credential
     :param delivery: A Delivery object
-    :return:
+    :return [str]: list of dds_ids of users we were unable to share project with after successful transfer
```

Comment:
It's unexpected that `accept_delivery` now returns this list of failed share users, given the name. I think it's cleaner to have the sharing with additional users be done by second function after `accept_delivery`.

Or, since the logic is now encapsulated in `DeliveryUtil`, maybe we don't need `accept_delivery` at all. The view can just create a `DeliveryUtil` and call its two methods.